hoomd/hpmc/tune/move_size.py

```diff
@@ -0,0 +1,333 @@
+        # not a performance problem.
+        self._tuned = 0
+        self._attached = False
+        # This is a bit complicated because we are having to ensure that we keep
+        # the list of tunables and the solver updated with the changes to
+        # attributes. However, these are simply forwarding a change along.
+        param_dict = ParameterDict(
+            moves=OnlyIf(to_type_converter([OnlyFrom(['a', 'd'])]),
+                         postprocess=update_moves),
+            types=OnlyIf(to_type_converter([str]),
+                         postprocess=update_types,
+                         allow_none=True),
+            target=OnlyType(float, postprocess=target_postprocess),
+            max_move_size=OnlyType(
+                float, allow_none=True,
+                postprocess=max_move_size_postprocess
+                ),
+            solver=SolverStep
+            )
+
+        param_dict['moves'] = moves
+        param_dict['types'] = types
+        param_dict['max_move_size'] = max_move_size
+        param_dict['target'] = target
+        param_dict['solver'] = solver
+        self._param_dict.update(param_dict)
+        if types is not None:
+            self._update_tunables(new_moves=moves, new_types=types)
+
+    def attach(self, simulation):
+        if not isinstance(simulation.operations.integrator, _HPMCIntegrator):
+            raise RuntimeError(
+                "MoveSizeTuner can only be used in HPMC simulations.")
+        particle_types = simulation.state.particle_types
+        if self.types is None:
+            self.types = particle_types
+        if not all(t in particle_types for t in self.types):
+            raise RuntimeError(
+                "Invalid particle type found specified types for tuning.")
+        self._update_tunables(new_moves=self.moves, new_types=self.types)
+        self._update_tunables_attr(
+            'integrator', simulation.operations.integrator)
+        self._attached = True
+
+    @property
+    def is_attached(self):
+        """bool: Whether or not the tuner is attached to a simulation."""
+        return self._attached
+
+    @property
+    def tuned(self):
+        """bool: Whether or not the move sizes are considered tuned.
+
+        Can be set to False.
+        """
+        return self._tuned == 2
+
+    @tuned.setter
+    def tuned(self, value):
+        if not value:
+            self._tuned = 0
+        else:
+            raise ValueError("Cannot attempt to set tuned to True. "
+                             "Detach operation instead.")
+
+    def detach(self):
+        self._update_tunables_attr('integrator', None)
+        self._attached = False
+
+    def act(self, timestep=None):
+        """Tune move sizes.
+
+        Args:
+            timestep (:obj:`int`, optional): Current simulation timestep. Is
+                currently ignored.
+        """
+        if not self.tuned and self.is_attached:
```

Comment:
This appears to make the tuner stop adjusting move sizes after it achieves its target. What if the conditions of the simulation change (i.e. the box size, or particle parameters) that cause this tuning to be invalid? Wouldn't we want to keep tuning?

Alternately, if we want to enforce tuning to stop the run when complete we should tie this tolerance check in with the new completion framework on the `quick-compress` branch. This way, users need to opt in to re-tuning. However, it would be less convenient to use:
```
sim.operations.tuners.append(move_size_tuner)
sim.run(1e6)
# stops early
sim.operations.tuners.remove(move_size_tuner)
sim.run(1e6)
```

Reply:
That is true. `tuned` is a settable property, though so it can be set to false to force retuning. We should mix this with the completion framework as well. I think the `tuned` property fits into this. It is completed when it is `tuned`, but that can be reset if desired; for instance, if the system changes.